### PR TITLE
Data sources and generic widgets

### PR DESCRIPTION
**Pull request by artemstepanenko**

### CODE REVIEW

File: src/containers/Dashboard.js [hunk 1]
```diff
@@ -13,15 +13,12 @@ class Dashboard extends Component {
   }
 
   render() {
+    // const customWidgets = [{ key: 'customWidget', type: 'vert-chart', url: 'http://www.mocky.io/v2/5968ca17110000d30d614c4d' }];
```
Comment (gkatsanos) [inline src/containers/Dashboard.js]: comment

File: src/lib/fetchApi.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import 'whatwg-fetch';
 
-const fetchApi = endpoint =>
-  fetch(`/api/${endpoint}`).then(response => response.json());
+const fetchApi = endpoint => fetch(`/api/${endpoint}`).then(response => response.json());
+
+export const fetchUrl = url => fetch(url).then(response => response.json());
```
Comment (gkatsanos) [inline src/lib/fetchApi.js]: not a blocker for now but I think having two methods just to not retype "api" is a bit exaggerated/over-engineered... I suggest either simply removing "api" from the server endpoint url .. or merge the two methods and include "api" in the url manually.. (I prefer the 1st solution :D )   @milenvoutchev

**Comment by gkatsanos** [comment]
very good work guys ! I'm amazed at how much progress was done and how nicely the code is abstracted!  It can be merged for me, but keep in mind the two small comments

File: src/containers/Dashboard.js (hunk 1 above)
Comment (artemstepanenko) [inline src/containers/Dashboard.js]: Removed.

**Comment by milenvoutchev** [comment]
Nice 👍

**Pull request merged**